README.md

```diff
@@ -4,690 +4,214 @@
-the property which has an unexpected value.
+## harness.expect
 
-*Note:* The type of the return value can often not be valid for the property value that you are passing it for.  You may need to cast
-it as `any` in order to allow TypeScript type checking to succeed.
-
-An example of usage would be:
+The most common requirement for testing is to assert the structural output from a widget's `render` function. `expect` accepts a render function that returns the expected render output from the widget under test.
 
 ```ts
-import { compareProperty } from '@dojo/test-extras/support/d';
-
-const compareRegistryProperty = compareProperty((value) => {
-    return value instanceof Registry;
-});
-
-widget.expectRender(v('div', {}, [
-    w('child', { registry: compareRegistryProperty })
+h.expect(() => v('div', { key: 'foo'}, [
+    w(Widget, { key: 'child-widget' }),
+    'text node',
+    v('span', { classes: [ 'class' ] })
 ]));
 ```
 
-### findIndex()
+If the actual render output and expected render output are different an exception is thrown with a structured visualization indicating all differences with `(A)` (the actual value) and `(E)` (the expected value).
 
-Returns a node identified by the supplied `index`.  The first argument is the _root_ virtual DOM node (`WNode` or `HNode`) and the
-second argument is the `index` being searched for.  Indexes can be either numbers, or a string of comma delimited numbers which
-specify the deeper index.  For example a string of `0,1,2` would get the third child of the second child of the first child of the
-_root_.  If resolved, the function will return the `DNode`, otherwise it returns `undefined`.
+Example assertion failure output:
 
-An example:
-
-```typescript
-const vdom = const expected = v('div', [
-    v('ol', { type: 'I' }, [
-        v('li', { value: 3 }, [ 'foo' ]),
-        v('li', { }, [ 'bar' ]),
-        v('li', { }, [ 'baz' ])
-    ])
-]);
-
-findIndex(vdom, '0,0,0'); // returns 'foo'
-```
-
-### findKey()
-
-Returns a node identified by the supplied `key`.  The first argument is the _root_ virutal DOM node (`WNode` or `HNode`) and the
-second argument is the `key` being searched for.  If found, the function will return the `WNode` or `HNode`, otherwise it returns
-`undefined`.
-
-An example:
-
-```typescript
-const vdom = const expected = v('div', [
-    v('ol', { type: 'I' }, [
-        v('li', { key: 'foo' }, [ 'foo' ]),
-        v('li', { }, [ 'bar' ]),
-        v('li', { }, [ 'baz' ])
-    ])
-]);
+```ts
+v("div", {
+	"classes": [
+		"root",
+(A)		"other"
+(E)		"another"
+	],
+	"onclick": "function"
+}, [
+	v("span", {
+		"classes": "span",
+		"id": "random-id",
+		"key": "label",
+		"onclick": "function",
+		"style": "width: 100px"
+	}, [
+		"hello 0"
+	])
+	w(ChildWidget, {
+		"id": "random-id",
+		"key": "widget"
+	})
+	w("registry-item", {
+		"id": true,
+		"key": "registry"
+	})
+])
+```
+
+### harness.expectPartial
```

Comment:
should things like this harness.expectPartial be `harness.expectPartial` ? (If not, ignore)